src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -272,20 +308,25 @@ internal virtual void Validate()
             // Validate that we have a client id
             if (string.IsNullOrWhiteSpace(Config.ClientId))
             {
-                throw new InvalidOperationException(MsalErrorMessage.NoClientIdWasSpecified);
+                throw new MsalClientException("no_client_id", MsalErrorMessage.NoClientIdWasSpecified);
```

Comment:
is the "no_client_id" the MsalErrorCode?  If so, shouldn't this be a constant in MsalErrorCode.cs?

Reply:
That's what I get from trying to be nice :D :D . I'll fix this.